app/lib/features/space/providers/space_navbar_provider.dart

```diff
@@ -1,157 +1,103 @@
-              color,
-              BlendMode.srcIn,
-            ),
-          ),
-          target: Routes.spaceTasks,
-        ),
-      );
+      final taskList = await ref.watch(spaceTasksListsProvider(spaceId).future);
+      if (taskList.isNotEmpty) {
+        tabs.add(
+          const TabEntry(key: TabEntry.tasks, label: 'Tasks'),
+        );
+      }
     }
 
     if (appSettings.events().active()) {
-      tabs.add(
-        TabEntry(
-          key: TabEntry.events,
-          label: 'Events',
-          makeIcon: (ctx, color) => Icon(
-            Atlas.calendar_schedule_thin,
-            color: color,
-          ),
-          target: Routes.spaceEvents,
-        ),
-      );
+      final eventList = await ref.watch(spaceEventsProvider(spaceId).future);
+      if (eventList.isNotEmpty) {
+        tabs.add(
+          const TabEntry(key: TabEntry.events, label: 'Events'),
+        );
+      }
     }
   }
 
-  tabs.add(
-    TabEntry(
-      key: TabEntry.chatsKey,
-      label: 'Chats',
-      makeIcon: (ctx, color) => Icon(
-        Atlas.chats_thin,
-        color: color,
-      ),
-      target: Routes.spaceChats,
-    ),
-  );
+  final spacesList = await ref.watch(relatedSpacesProvider(spaceId).future);
+  if (spacesList.isNotEmpty) {
+    tabs.add(
+      const TabEntry(key: TabEntry.spacesKey, label: 'Spaces'),
+    );
+  }
 
-  tabs.add(
-    TabEntry(
-      key: const Key('spaces'),
-      label: 'Spaces',
-      makeIcon: (ctx, color) => Icon(
-        Atlas.connection_thin,
-        color: color,
-      ),
-      target: Routes.spaceRelatedSpaces,
-    ),
-  );
+  final chatsList = await ref.watch(relatedChatsProvider(spaceId).future);
+  if (chatsList.isNotEmpty) {
+    tabs.add(
+      const TabEntry(key: TabEntry.chatsKey, label: 'Chats'),
+    );
+  }
 
   tabs.add(
-    TabEntry(
-      key: const Key('members'),
-      label: 'Members',
-      makeIcon: (ctx, color) => Icon(
-        Atlas.group_team_collective_thin,
-        color: color,
-      ),
-      target: Routes.spaceMembers,
-    ),
+    const TabEntry(key: TabEntry.membersKey, label: 'Members'),
   );
-  return tabs;
-});
-
-class SelectedTabNotifier extends Notifier<Key> {
-  @override
-  Key build() {
-    return const Key('overview');
-  }
 
-  void switchTo(Key input) {
-    WidgetsBinding.instance.addPostFrameCallback((Duration duration) {
-      state = input;
-    });
+  final membership = ref.watch(roomMembershipProvider(spaceId));
+  bool canAddPin = membership.requireValue!.canString('CanPostPin');
+  bool canAddEvent = membership.requireValue!.canString('CanPostEvent');
+  bool canAddTask = membership.requireValue!.canString('CanPostTaskList');
+  bool canLinkSpaces = membership.requireValue!.canString('CanLinkSpaces');
+
+  //Show action menu only if you have at lease one permission
+  if (canAddPin | canAddEvent | canAddTask | canLinkSpaces) {
+    tabs.add(
+      const TabEntry(key: TabEntry.actionsKey, label: '•••'),
+    );
```

Comment:
This for temporarily, once we implemented on click activation of add pins, add event etc then this will no longer require as Action Menu always be there with at lease couple of action buttons.